### PR TITLE
docs_examples

### PR DESCRIPTION
**Pull request by tomasonjo**
Added a basic syntax for all:

Two things come to mind:
 - PageRank has scoreProperty and writeProperty that both default to "score", which is confusing
 - UnionFind has property parameter that should be a weightProperty i guess

### CODE REVIEW

**Comment by jexp** [comment]
@tomasonjo is this ready to merge?

**Comment by tomasonjo** [comment]
Yes it is fine to merge so that we show all the syntax for now. Need to correct some syntax in a different PR for more clarity next I think.

File: doc/connected-components.adoc [hunk 1]
```diff
@@ -12,10 +18,79 @@ _Connected Components_ or _UnionFind_ basically finds sets of connected nodes wh
...
+| relationship | string | null | yes | relationship-type to load from the graph, if null load all relationships
+
+|===
+
+.Results
+[opts="headers"]
+|===
+| name | type | description
+| nodeId | long | node id
+| setId | int | cluster id
```
Comment (jexp) [inline doc/connected-components.adoc]: there is no difference between long and `int` we should use `integer` and `float` (i.e. the cypher types) consistently throughout the docs.

File: doc/label-propagation.adoc [hunk 2]
```diff
@@ -1,22 +1,66 @@
...
+[opts="header",cols="1,1,6"]
+|===
+| name | type | description
+| nodes | int | number of nodes considered
+| iterations | int | number of iterations
+| loadMillis | int | milliseconds for loading data
+| computeMillis | int | milliseconds for running the algorithm
+| writeMillis | int | milliseconds for writing result data back
+| weightProperty | string | property name that contains weight
+| partitionProperty | string | property name written back to
```
Comment (jexp) [inline doc/label-propagation.adoc]: what is written back

File: doc/label-propagation.adoc [hunk 3]
```diff
@@ -1,22 +1,66 @@
...
+
+|===
+
+.Results
+[opts="header",cols="1,1,6"]
+|===
+| name | type | description
+| nodes | int | number of nodes considered
+| iterations | int | number of iterations
+| loadMillis | int | milliseconds for loading data
```
Comment (jexp) [inline doc/label-propagation.adoc]: include from common results table

File: doc/minimum-weight-spanning-tree.adoc [hunk 4]
```diff
@@ -1,24 +1,76 @@
...
+There are quite a few use cases for minimum spanning trees. One example would be a telecommunications company which is trying to lay out cables in new neighborhood. If it is constrained to bury the cable only along certain paths (e.g. along roads), then there would be a graph representing which points are connected by those paths. Some of those paths might be more expensive, because they are longer, or require the cable to be buried deeper; these paths would be represented by edges with larger weights. Currency is an acceptable unit for edge weight – there is no requirement for edge lengths to obey normal rules of geometry such as the triangle inequality. A spanning tree for that graph would be a subset of those paths that has no cycles but still connects to every house; there might be several spanning trees possible. A minimum spanning tree would be one with the lowest total cost, thus would represent the least expensive path for laying the cable.
+
 == Constraints / when not to use it
 
 == Algorithm explanation on simple sample graph
 
 [source,cypher]
 ----
+MATCH(n:Node{start:true}) 
+WITH n
```
Comment (jexp) [inline doc/minimum-weight-spanning-tree.adoc]: we don't need the `WITH n`

File: doc/minimum-weight-spanning-tree.adoc [hunk 5]
```diff
@@ -1,24 +1,76 @@
...
+
 == Constraints / when not to use it
 
 == Algorithm explanation on simple sample graph
 
 [source,cypher]
 ----
+MATCH(n:Node{start:true}) 
+WITH n 
+CALL algo.mst(n, 'cost', {write:true, stats:true,writeProperty:"mst"})
```
Comment (jexp) [inline doc/minimum-weight-spanning-tree.adoc]: do we need stats:true in the stats procedure?

File: doc/pagerank.adoc [hunk 6]
```diff
@@ -31,26 +34,33 @@ CALL algo.pageRank(label:String, relationship:String, {iterations:5, dampingFact
 | name | type | default | optional | description
 | label  | string | null | yes | label to load from the graph, if null load all nodes
 | relationship | string | null | yes | relationship-type to load from the graph, if null load all nodes
-| iterations | int | 5 | yes | how many iterations of page-rank to run
+| iterations | int | 20 | yes | how many iterations of page-rank to run
 | dampingFactor | float | 0.85 | yes | damping factor of the page-rank caculation
 | write | boolean | true | yes | if result should be written back as node property
-| scoreProperty | string | 'score' | yes | property name written back to
+| writeProperty | string | 'score' | yes | property name written back to
+| scoreProperty | string | 'score' | yes | property name that contains weight
```
Comment (jexp) [inline doc/pagerank.adoc]: should both scoreProperty and writeProperty have the same value? shouldn't writeProperty be sth like "rank" or "pageRank" ?

Comment (jexp) [inline doc/pagerank.adoc]: or "scoreProperty" should be "weigthProperty" or "propertyName" ?

File: doc/single-shortest-path.adoc [hunk 7]
```diff
@@ -14,6 +14,34 @@ _All Pairs Shortest Path_ on the other hand calculates a shortest path forest co
...
+----
+MATCH (start:Node{type:'start'}), (end:Node{type:'end'})
+CALL algo.shortestPath(start, end, 'cost') 
+YIELD loadDuration, evalDuration, nodeCount, totalCost
+
+----
+
+[source,cypher]
+----
+MATCH(n:Node {name:'s'}) WITH n CALL algo.deltaStepping.stream(n, 'cost', 3.0)
```
Comment (jexp) [inline doc/single-shortest-path.adoc]: @mknblch should we expose implementation details in the procedure name? Can we find the better "common" algorithm name that's covered here?

File: doc/strongly-connected-components.adoc [hunk 8]
```diff
@@ -12,10 +14,46 @@ _SCC_ is a class algorithms for finding groups of nodes where each node is direc
...
 ----
 
 == Example Usage
 
+== Syntax
+
+.Running algorithm and writing back results
+[source,cypher]
+----
+CALL algo.scc(label:String, relationship:String, {write:true,clusterProperty:'cluster'})
```
Comment (jexp) [inline doc/strongly-connected-components.adoc]: @mknblch missing writeProperty ? both as config and result?

Comment (tomasonjo) [inline doc/strongly-connected-components.adoc]: clusterProperty is the "writeProperty" in this scenario

File: doc/pagerank.adoc (hunk 6 above)
Comment (tomasonjo) [inline doc/pagerank.adoc]: Yes I think that it should be different defaults... i went with what I read from the code... I think that "scoreProperty" is "weightProperty" not sure... we should avoid using propertyName as it is confusing imo

File: doc/connected-components.adoc [hunk 9]
```diff
@@ -12,10 +18,79 @@ _Connected Components_ or _UnionFind_ basically finds sets of connected nodes wh
...
+
+----
+
+.Parameters
+[opts="header",cols="1,1,1,1,4"]
+|===
+| name | type | default | optional | description
+| label  | string | null | yes | label to load from the graph, if null load all nodes
+| relationship | string | null | yes | relationship-type to load from the graph, if null load all nodes
+| property | string | null | yes | property name that contains weight
```
Comment (tomasonjo) [inline doc/connected-components.adoc]: is there any difference in weight vs cost (I think not), both are properties of rels ? should just pick one and be consistent... avoid property as it is confusing

File: doc/label-propagation.adoc (hunk 2 above)
Comment (tomasonjo) [inline doc/label-propagation.adoc]: The partition of the graph in which the node resides

File: doc/connected-components.adoc [hunk 10]
```diff
@@ -12,10 +18,79 @@ _Connected Components_ or _UnionFind_ basically finds sets of connected nodes wh
...
 ----
 
 == Example Usage
 
+== Syntax
+
+.Running algorithm and writing back results
+[source,cypher]
+----
+CALL algo.unionFind(label:String, relationship:String, {property:'propertyName', threshold:0.42,
```
Comment (tomasonjo) [inline doc/connected-components.adoc]: should change also the `property` to `weightProperty`, just spoke with martin and he said that whenever the default value for `property` is cost or the name of the param is `costProperty` it means the same as `weightProperty`. So I would stick with `weightProperty`

Comment (jexp) [inline doc/connected-components.adoc]: Ok, makes sense, as long as we are consistent, and it makes sense for the user.

File: doc/connected-components.adoc (hunk 9 above)
Comment (jexp) [inline doc/connected-components.adoc]: Yes. Can you work with @mknblch  on this?

File: doc/label-propagation.adoc (hunk 2 above)
Comment (jexp) [inline doc/label-propagation.adoc]: Put it in :)

File: doc/minimum-weight-spanning-tree.adoc (hunk 5 above)
Comment (jexp) [inline doc/minimum-weight-spanning-tree.adoc]: I mean generally.

File: doc/pagerank.adoc (hunk 6 above)
Comment (jexp) [inline doc/pagerank.adoc]: Please ask @mknblch

File: doc/strongly-connected-components.adoc (hunk 8 above)
Comment (jexp) [inline doc/strongly-connected-components.adoc]: Then let's change it.

**Comment by jexp** [comment]
@tomasonjo should I merge this in and then you work on my comments in another round?

**Comment by tomasonjo** [comment]
No please do not merge. Will push commits today afternoon.

**Comment by tomasonjo** [comment]
added some examples and updated the syntax according to [#160](https://github.com/neo4j-contrib/neo4j-graph-algorithms/issues/160)

**Pull request merged**